slim/datasets/cifar10.py

```diff
@@ -93,6 +93,4 @@ def get_split(split_name, dataset_dir, file_pattern=None, reader=None):
       reader=reader,
       decoder=decoder,
       num_samples=SPLITS_TO_SIZES[split_name],
-      items_to_descriptions=_ITEMS_TO_DESCRIPTIONS,
-      num_classes=_NUM_CLASSES,
-      labels_to_names=labels_to_names)
+      items_to_descriptions=_ITEMS_TO_DESCRIPTIONS)
```

Comment:
The first error @vonclites hit is just because those two fields (`num_classes` and `labels_to_names`) are deleted by this patch. `Dataset` accepts `**kwargs` where you can put any name/value pairs you want.

`eval_image_classifier.py` is using `num_classes` to initialize the various network's output layer. 

The networks all have a default number of classes, but you need the dataset's number of classes if you apply a network to a different dataset (like using cifarnet on cifar10 or cifar100).